app/models/concerns/disabling.rb

```diff
@@ -0,0 +1,37 @@
+# The disposable module is a soft-delete helper that:
+#
+#  * adds `disable!` method that set a `disabled_at` attribute anden then _buries_ the object
+#  * adds a `bury!`  hook method that allows further modification when disabling
+#  * aliases `destroy!` and `destroy` to `disable!`, but still keeps `delete` and friends
+#
+module Disabling
+  extend ActiveSupport::Concern
+
+  def disable!
+    transaction do
+      update_attribute :disabled_at, Time.current
+      bury!
+    end
+  end
+
+  def disabled?
+    disabled_at.present?
+  end
+
+  def enabled?
+    !disabled?
+  end
+
+  # override to perform additional
+  # post-disable actions
+  def bury!
+  end
+
+  def ensure_enabled!
+    raise Mumuki::Domain::ForbiddenError if disabled?
```

Comment:
In some contexts I think a NotFound is more accurate but I'm assuming that we are using this to represent that it's beyond user's scope, so :+1:

Reply:
I actually would like to use a new custom exception, `Mumuki::Domain::DisabledResourceError`, that would generate a 503 error in laboratory. Does it make sense?

Reply:
Totally!